### PR TITLE
Delete test files generated from running sphinx gallery examples

### PR DESCRIPTION
**Pull request by rly**
## Motivation

Running the Sphinx gallery examples locally (`python test.py -e`) generates a lot of test files that are frustrating to delete. This PR deletes the resulting test files after running those examples in the test suite. In order to delete the test files on Windows, the h5py file handle must be explicitly closed, so this PR also updates the Sphinx gallery examples to close the file.

## How to test the behavior?
```
python test.py -e
```

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -228,6 +231,49 @@ def run_integration_tests(verbose=True):
     run_test_suite("tests/validation", "validation CLI tests", verbose=verbose)
 
 
+def clean_up_tests():
+    # remove files generated from running example files
```
Comment (oruebel) [inline test.py:235]: Can you add a bullet item in the "Create a new tutorial" section of https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/source/make_a_tutorial.rst to mention that any files generated by the tutorial should be add to this list.

**Review by oruebel — CHANGES_REQUESTED** [review]
Aside from updating the instructions for creating tutorials this looks good to me.

File: test.py (hunk 1 above)
Comment (rly) [reply test.py:235]: Done

**Review by oruebel — DISMISSED** [review]

**Comment by rly** [comment]
Code coverage for the validation script was not being maintained and uploaded to codecov. I fixed that. Please re-approve

**Review by oruebel — APPROVED** [review]

**Comment by oruebel** [comment]
Looks good. Thanks for addressing this issue.

**Pull request merged**